### ISSUE TITLE
Sorry to bother, have you succeed in passing the IGD?

### ISSUE THREAD

**Issue by NyaMisty** [lansing/osx-kvm-igd#1]
Could the scripts in this repo boot up the MacOS?

**Comment by lansing** [comment]
@NyaMisty Yeah I was successful booting the OS using the IGD passed through. However there were sporadic crashes under certain applications, related to OpenGL. This was confirmed by other people as well, and we were not able to find a fix. I recommend using AMD GPU passthrough and using the IGD for linux.

**Comment by NyaMisty** [comment]
Wow thanks for such a quick response.
Well in fact I'm using a laptop with MUXless Optimus implementation, so passing the iGPU is my only choice :(
Could you please tell me how to install the Mojave on Seabios and i440fx? Seems that people are all using Clover + OVMF. Do you use the Eroch or is there any trick?

**Comment by lansing** [comment]
To be honest I don't remember the initial install that well, it was over a year ago. However, I am pretty sure that I actually installed it using the emulated graphics. I definitely used Clover. I'd recommend starting with zero passthrough devices, i.e. emulated graphics and network, and get a working system that way. Then try passing through the iGPU. You probably need to be booting linux entirely on the discrete GPU, i.e. you do not want so much as a framebuffer driver hitting the iGPU before you start the VM.

Do note that the crashes were severe and were happening at least once per day. A lot of applications use OpenGL that you wouldn't expect, simply for accelerated 2D. I'm a programmer and a all of my IDEs were crashing the iGPU environment daily. Not worth the trouble IMO but good luck, and let me know if you end up with a stable system.

**Comment by NyaMisty** [comment]
Got it. Well I've got some

**Comment by NyaMisty** [comment]
Got it. I've searched for a while and it seems that we need to customize some parameter in Clover to avoid random crashes: https://www.elitemacx86.com/threads/fix-intel-hd-graphics-630-on-macos-10-12-6-and-later.216/

**Comment by NyaMisty** [comment]
Also could you please give me your device information? Is it a laptop?
Thanks a lot!

**Comment by lansing** [comment]
Not to be a naysayer but I've tried all this stuff. The crashes that I and others encountered doing passthrough were completely different from the crashes people encounter when using iGPU on bare metal. My machine was (and still is) perfectly stable using iGPU on bare metal. Several people have come to me saying they googled some workaround and then it turns out not to be helpful in the passthrough context. I tried every version of QEMU I could get my hands on, different linux kernels, every possible QEMU param, every permutation of clover configs, etc etc etc. Again, not trying to dissuade you if you truly believe you can beat this and have a lot of time on your hands to try, just sharing my experience. And if you are successful, I will be super stoked, because it means I can free up a PCIe slot, so please let me know.

My machine is a desktop, 8700k with z390 chipset. Only reason I wanted to passthrough iGPU was so I could free up a slot for another nvidia GPU, as this is a machine learning workstation. 

There is a

**Comment by NyaMisty** [comment]
Thanks for so much information you give me. For me I'm not always using the passthrough, I'm trying to run the macOS from both VMware on Windows and QEMU. I'm only switching to the QEMU when I need to do some usb debugging stuff.
Also I'll try to cope with that issue if it really bothers a lot.

**Comment by NyaMisty** [comment]
Ah, could you also tell me whether your vbios is the original one?
Did you patch it?

**Comment by lansing** [comment]
`vbios.dump` is the raw dump I took. `vbios-fixed.dump` is with some patch applied. I don't remember what exactly the patch was, I must have been just following some instructions I found. As I recall I tried both versions and didn't see any difference.

One more bit of general advice here. If you have not gotten IGD passthrough for macOS working yet, it's probably easier to work on booting a windows VM with legacy IGD passthrough. The windows intel graphics drivers are much more forgiving. Then at least you know your host and QEMU is configured right, and can move onto getting macOS to boot.
